src/flexalloc_mm.c

```diff
@@ -1722,8 +1722,19 @@ fla_open(struct fla_open_opts *opts, struct flexalloc **fs)
     goto free_dev_uri;
   }
 
+  if(md_dev != dev)
+  {
+    (*fs)->dev.md_dev_uri = strdup(opts->md_dev_uri);
+    if ((*fs)->dev.md_dev_uri == NULL)
+    {
+      err = -ENOMEM;
+      goto free_dev_uri;
```

Comment:
If this fails the allocation at line 1718 (opts->dev_uri) needs to be freed also, so you'll need a new label above free_dev_uri to handle this case.